.github/workflows/ci.yml

```diff
@@ -0,0 +1,41 @@
+name: test
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.15, 1.19, 1.22]
+
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Build
+      run: |
+        make
+
+    - name: Run tests
+      run: |
+        make test
+
+    - name: Create release tarball
```

Comment:
Why not make that only on tags?

Reply:
So we test that it works and continues to work

Reply:
Also note that building the release tarball and uploading it takes 2 seconds in CI.